torchbenchmark/score/compute_score.py

```diff
@@ -7,148 +7,151 @@
-
-    score = target * math.exp(score)
-    return score
+class TorchBenchScore:
+    def __init__(self, spec=SPEC_FILE_DEFAULT, target=TARGET_SCORE_DEFAULT):
+        self.spec = spec
+        self.target = target
+        self.weights = None
+        self.norm = None
+
+    def setup_weights(self):
+        """
+        Calculates the static benchmark weights by iterating the spec
+        file and constructs a dictionary with (key, value) pair
+        is (task, weight_for_benchmark_per_task)
+
+        The spec here can be user defined .yaml file or if no spec is defined,
+        the default score.yaml is used.
+        """
+        # Load the spec file
+        with open(self.spec) as spec_file:
+            spec = yaml.full_load(spec_file)
+
+        self.weights = defaultdict(float)
+        category_spec = spec['hierarchy']['model']
+        domain_weight = 1.0/ len(category_spec)
+        for domain in category_spec:
+            tasks = category_spec[domain]
+            task_weight = 1.0 / len(tasks)
+            for task in tasks:
+                benchmarks = tasks[task]
+                benchmark_weight = 1.0 / len(benchmarks)
+                self.weights[task] = domain_weight * task_weight * benchmark_weight
+
+    def setup_benchmark_norms(self, data):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+
+        This reference data can be user provided or by default the first file
+        in the directory of data files is considered as the reference data file
+        """
+        self.norm = {b['name']: b['stats']['mean'] for b in data['benchmarks']}
+
+    def _get_model_task(self, model_name):
+        """
+        Helper function which extracts the task the model belongs to
+        by iterating over the Model attributes.
+        """
+        MODEL_PATH="../../models"
+        p = Path(__file__).parent.joinpath(MODEL_PATH + model_name)
+        if(p):
+            module = importlib.import_module(f'torchbenchmark.models.{model_name}', package=__name__)
+            Model = getattr(module, 'Model')
+        return Model.task.value
+
+    def _get_benchmark_configuration(self, b):
+        """
+        This helper function extracts the configuration details
+        from the benchmark name.
+        For eg., if benchmark_name = test_eval[pytorch_struct-cuda-jit]
+                test, device, mode = eval, cuda, jit
+                model_name = pytorch_struct
+
+        This helper also extracts the task and model_name of the benchmark
+        The function returns a tuple of the configs, model task, model_name
+        """
+        test, device, mode = "train", "cuda", "eager"
+        if "eval" in b: test = "eval"
+        if "cpu" in b: device = "cpu"
+        if "jit" in b: mode = "jit"
+
+        # Extract the model name from the benchmark b
+        model_name = (re.findall(r'\[(.*)\-'+device, b))[0]
```

Comment:
for this i was thinking you could do it all in one line and not even need this helper function.  e.g.

test, device, mode = re.parse("the regexp", b).groups()[0]

well, it would still need error handling before doing [0] but that's the only thing.  
https://regex101.com/ is good if you want to quickly figure out the regexp.  i would use - and [ characters and begin-line character as anchors and () to grab the different parts.

Reply:
Ok, done.